src/guest/vm_flash.cc

```diff
@@ -66,66 +66,74 @@ bool VmFlasher::QemuCreateVirtUsbDisk(void) {
-               " count=" + std::to_string((total_image_size_ + 1_GB + kDdBs - 1)/kDdBs));
-    LOG(info) << cmd;
-    if (boost::process::system(cmd)) {
-        LOG(error) << "Failed to : " << cmd;
-        return false;
-    }
+        boost::filesystem::path boost_file(o_dir.string() + "/boot.img");
+        if (boost::filesystem::exists(boost_file)) {
+            if (!CheckImages(o_dir))
+            return false;
 
-    cmd.assign("mkfs.vfat " + std::string(kVirtualUsbDiskPath));
-    LOG(info) << cmd;
-    if (boost::process::system(cmd)) {
-        LOG(error) << "Failed to : " << cmd;
-        return false;
-    }
+            cmd.assign("dd if=/dev/zero of=" + std::string(kVirtualUsbDiskPath) +
+                    " bs=" + std::to_string(kDdBs) +
+                    " count=" + std::to_string((total_image_size_ + 1_GB + kDdBs - 1)/kDdBs));
+            LOG(info) << cmd;
+            if (boost::process::system(cmd)) {
+                LOG(error) << "Failed to : " << cmd;
+                return false;
+            }
 
-    boost::filesystem::directory_iterator end_itr;
-    for (boost::filesystem::directory_iterator ditr(o_dir); ditr != end_itr; ++ditr) {
-        if (boost::filesystem::is_regular_file(ditr->path(), bec)) {
-            cmd.assign("mcopy -o -n -i " + std::string(kVirtualUsbDiskPath) + " " + ditr->path().string() + " ::");
+            cmd.assign("mkfs.vfat " + std::string(kVirtualUsbDiskPath));
             LOG(info) << cmd;
             if (boost::process::system(cmd)) {
                 LOG(error) << "Failed to : " << cmd;
                 return false;
             }
-        }
-    }
 
-    cmd.assign("rm -r " + o_dir.string());
-    LOG(info) << cmd;
-    if (boost::process::system(cmd)) {
-        LOG(warning) << "Failed to : " << cmd;
-    }
+            boost::filesystem::directory_iterator end_itr;
+            for (boost::filesystem::directory_iterator ditr(o_dir); ditr != end_itr; ++ditr) {
+                if (boost::filesystem::is_regular_file(ditr->path(), bec)) {
+                    cmd.assign("mcopy -o -n -i " + std::string(kVirtualUsbDiskPath) + " " + ditr->path().string() + " ::");
+                    LOG(info) << cmd;
+                    if (boost::process::system(cmd)) {
+                        LOG(error) << "Failed to : " << cmd;
+                        return false;
+                    }
+                }
+            }
+            
+            cmd.assign("rm -r " + o_dir.string());
+            LOG(info) << cmd;
+            if (boost::process::system(cmd)) {
+                LOG(warning) << "Failed to : " << cmd;
+            }
+
+            virtual_disk_ = kVirtualUsbDiskPath;
 
+        } else {
+            boost::filesystem::directory_iterator end_itr;
+            for (boost::filesystem::directory_iterator ditr(o_dir); ditr != end_itr; ++ditr) {
+                virtual_disk_ = ditr->path().string();
+            }
+        }
+    } else {
```

Comment:
the if statement seems a little bit deep. Could you try implement as below pesudo code?

```
if (not zip file) {
    set virtual_disk_;
    return true;
}

unzip zip file;
if (only one regular file in o_dir) {
    set virtual_disk_;
    return true;
}

if (boot.img in o_dir) {
    create vusb disk;
    return true;
}

return false;
```